velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -3498,6 +3498,29 @@ TEST_F(DateTimeFunctionsTest, timeZoneMinute) {
       "Unable to parse timestamp value: \"2023-\", expected format is (YYYY-MM-DD HH:MM:SS[.MS])");
 }
 
+TEST_F(DateTimeFunctionsTest, fromISO8601FunctionDate) {
+  const auto fromISODate =
+      [&](const std::optional<std::string>& isoDateString) {
+        return evaluateOnce<int32_t>("from_iso8601_date(c0)", isoDateString);
+      };
+
+  EXPECT_EQ(0, fromISODate("1970-01-01T03:19:58.000"));
+  EXPECT_EQ(18297, fromISODate("2020-02-05T14:27:39.000-07:00"));
+  EXPECT_EQ(-18297, fromISODate("1919-11-28T23:59:59.999"));
+  EXPECT_EQ(-18297, fromISODate("1919-11-28T23:59:59.999+08:00"));
+  EXPECT_EQ(-2, fromISODate("1969-12-30T19:00:00.000-05:00"));
+  EXPECT_EQ(-719528, fromISODate("0000-01-01T00:00:00.000+00:00"));
+  EXPECT_EQ(2932896, fromISODate("9999-12-31:23:59.999+00:00"));
```

Comment:
Why this one is passing?

Reply:
Sorry, I don't understand - is there a particular test there you are referring to? My general idea was that `from_iso8601_date` should convert an ISO-formatted timestamp string to a Date type, and as the Date type is nothing but a 32-bit integer now, representing number of days since the epoch, this should be a sufficient check of the validity of the function.

Reply:
Hmm, I tested it out and it seems that Java implementation of `from_iso8601_date()` is rejecting any ISO-formatted string:

- generating ISO-formatted string from `to_ISO8601()`
```
presto:default> SELECT to_iso8601(TIMESTAMP '2001-08-22 03:04:05.321 Africa/Cairo');
SELECT to_iso8601(TIMESTAMP '2001-08-22 03:04:05.321 Africa/Cairo');
             _col0             
-------------------------------
 2001-08-22T03:04:05.321+03:00 
(1 row)
```
- passing the above output string into `from_iso8601_date()`
```
presto:default> SELECT from_iso8601_date('2001-08-22T03:04:05.321+03:00');
SELECT from_iso8601_date('2001-08-22T03:04:05.321+03:00');
Query 20231027_232729_00005_bta8n failed: Invalid format: "2001-08-22T03:04:05.321+03:00" is malformed at "T03:04:05.321+03:00"
```

Possible bug?

Reply:
I think java is using ' ' instead 'T' here.  Can you match the behavior in Java?